mongo/database.go

```diff
@@ -103,10 +103,12 @@ func (db *Database) Collection(name string, opts ...*options.CollectionOptions)
 // Aggregate performs an aggregate operation (https://docs.mongodb.com/manual/reference/command/aggregate/) against
 // the database. This requires MongoDB version >= 3.6 and driver version >= 1.1.0.
 //
-// The pipeline parameter should be an array of documents, each representing an aggregation stage. The pipeline
-// or any of the stage documents cannot be nil. For a pipeline of bson.D documents, the mongo.Pipeline type can be used.
-// See https://docs.mongodb.com/manual/reference/operator/aggregation-pipeline/#db-aggregate-stages for a list of valid
-// stages in database-level aggregations.
+// The pipeline parameter should be an slice of documents, each representing an aggregation stage. The pipeline
+// or any of the stage documents cannot be nil. It can be empty. For a pipeline of bson.D documents, the mongo.Pipeline
```

Comment:
Unfortunately, it's ambiguous what "it" refers to in "It can be empty", since the preceding sentence's subject is compound with an "or".  Perhaps:

"The pipeline must be non-nil but can be empty. The stage documents must all be non-nil."

Reply:
good point. i didn't realize the ambiguity at first but i like the suggested phrasing. changed.